mongo_proxy/durable_cursor.py

```diff
@@ -227,9 +235,8 @@ def try_reconnect(self, get_next=True):
         raise MongoReconnectFailure()
 
     def count(self, with_limit_and_skip=False):
-        while True:
-            try:
-                return self.cursor.count(
-                    with_limit_and_skip=with_limit_and_skip)
-            except AutoReconnect:
-                self.try_reconnect(get_next=False)
+        return self._with_retry(
+            False,
```

Comment:
Passing the function to call as a keyword arg, e.g. `self._with_retry(get_next=False, f=self.cursor.count, ...)`, seemed a bit verbose, and swapping their order seemed confusing, as `get_next` is not an argument to `f`.

But I can get behind the verbosity, if it makes this clearer.